inc/class-controller.php

```diff
@@ -92,6 +92,52 @@ public function contentItemSubmit( $params ) {
 		echo $page;
 	}
 
+	/**
+	 * Creates a book from an LTI request
+	 *
+	 * @param $action
+	 * @param $params
+	 * @since 1.4.0
+	 */
+	public function createBook( $action, $params ) {
+		$connector = Database::getConnector();
+		$tool      = new Tool( $connector );
+		$tool->setAdmin( $this->admin );
+		$tool->setAction( $action );
+		$tool->processRequest( $params );
+		$tool->handleRequest();
+
+		$activity_url = $tool->buildAndValidateUrl( $_POST['resource_link_title'] );
+		$exists       = $tool->validateLtiBookExists( $activity_url, $_POST['resource_link_id'], $_POST['context_id'] );
+
+		if ( $exists || $tool->user->isLearner() ) {
+			\Pressbooks\Redirect\location( $activity_url );
+			do_exit();
+		}
+
+		if ( empty( $_SESSION['pb_lti_consumer_pk'] ) || empty( $_SESSION['pb_lti_consumer_version'] ) || empty( $_SESSION['pb_lti_return_url'] ) ) {
+			wp_die( __( 'You do not have permission to do that.' ) );
+		}
+
+		$new_book_url = $tool->maybeDisambiguateDomain( $activity_url );
+		$title        = $tool->buildTitle( $_POST['context_label'], $_POST['context_id'], $_POST['resource_link_title'], $_POST['resource_link_id'] );
+
+		$lti_id = "{$tool->consumer->consumerGuid}|{$tool->user->getId()}";
+		$wp_user = $tool->matchUserById( $lti_id );
+
+		if ( $wp_user ) {
```

Comment:
two `if` can be flatten in a single one:

`if ( ( $wp_user ) && ( $tool->user->isStaff() || $tool->user->isAdmin() )`

Reply:
complete!